src/main/java/de/retest/web/WebData.java

```diff
@@ -62,22 +62,28 @@ public Rectangle getOutline() {
 	}
 
 	/**
-	 * @throws NumberFormatException
-	 *             if the given value cannot be converted to integer.
+	 * @throws IllegalStateException
+	 *             if the value for the given key cannot be converted to integer.
 	 */
-	private int toInt( final Object value ) throws NumberFormatException {
-		if ( value instanceof Integer ) {
-			return (Integer) value;
-		}
-		if ( value instanceof String ) {
-			return Integer.parseInt( (String) value );
-		}
-		if ( value instanceof Double ) {
-			return Math.toIntExact( Math.round( (Double) value ) );
-		}
-		if ( value instanceof Long ) {
-			return Math.toIntExact( Math.round( (Long) value ) );
+	public int getAsInt( final String key ) {
+		final Object value = wrappedData.get( key );
+		try {
+			if ( value instanceof Integer ) {
+				return (Integer) value;
+			}
+			if ( value instanceof String ) {
+				return Integer.parseInt( (String) value );
+			}
+			if ( value instanceof Double ) {
+				return Math.toIntExact( Math.round( (Double) value ) );
+			}
+			if ( value instanceof Long ) {
+				return Math.toIntExact( Math.round( (Long) value ) );
+			}
+		} catch ( final Exception e ) {
+			// muted, since we throw a general exception below
 		}
-		throw new IllegalArgumentException( "Don't know how to convert a " + value.getClass() + " to int!" );
+		throw new IllegalStateException( "Don't know how to convert " + value + " of "
```

Comment:
Why changing to `IllegalStateException` this is no state. Calling upon a later point will still result in the same exception.

Reply:
The error depends not on the given input alone, but also on the contents of the webData, hence state.

Reply:
>Signals that a method has been invoked at an illegal or inappropriate **time**. In other words, the Java environment or Java application is not in an appropriate state for the requested operation.

Reply:
Why just not an own exeption...